MobileCommerceApp/app/src/main/java/shuvalov/nikita/mobilecommerceapp/offline_store_front/OfflineStoreActivity.java

```diff
@@ -0,0 +1,213 @@
+    int mGridColumns;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_offline_store);
+
+        Toolbar toolbar = (Toolbar) findViewById(R.id.tool_bar);
+        setSupportActionBar(toolbar);
+        mRecyclerView = (RecyclerView) findViewById(R.id.offline_store_recycler);
+
+
+        DbUpdateImagesAsyncTask updateImgTask = new DbUpdateImagesAsyncTask(this);
+        updateImgTask.execute();
+    }
+
+
+    public void updateDatabaseImageReferences(){
+        OfflineSQLOpenHelper dbHelper = OfflineSQLOpenHelper.getMyInstance(this);
+        ArrayList<Product> products = dbHelper.getInventoryAsList();
+
+        for(Product product: products){
+            int imageRef;
+            switch(product.getName()){
+                case "Ugly X-Mas Sweater":
+                    imageRef = R.drawable.ugly_xmas_sweater;
+                    break;
+                case "Holy Grail":
+                    imageRef = R.drawable.holy_grail;
+                    break;
+                case "Bottled Lightning":
+                    imageRef = R.drawable.bottled_lightning;
+                    break;
+                case "Monkey Paw":
+                    imageRef = R.drawable.monkey_paw;
+                    break;
+                case "Unconvincing Toupee":
+                    imageRef = R.drawable.unconvincing_toupee;
+                    break;
+                case "Denim Chicken":
+                    imageRef = R.drawable.denim_chicken;
+                    break;
+                case "Egg":
+                    imageRef = R.drawable.egg;
+                    break;
+                case "Rum Ham":
+                    imageRef = R.drawable.rumham;
+                    break;
+                case "Snake Oil":
+                    imageRef = R.drawable.snake_oil;
+                    break;
+                default:
+                    imageRef = 0;
+                    break;
+            }
+            dbHelper.updateImageReference(product.getName(),imageRef);
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater menuInflater = this.getMenuInflater();
+        menuInflater.inflate(R.menu.offline_menu, menu);
+        SearchManager searchManager = (SearchManager) getSystemService(Context.SEARCH_SERVICE);
+        SearchView searchView = (SearchView)menu.findItem(R.id.search).getActionView();
+        ComponentName componentName = new ComponentName(this, OfflineStoreActivity.class);
+        searchView.setSearchableInfo(searchManager.getSearchableInfo(componentName));
+        return true;
+    }
+
+
+    @Override
+    protected void onNewIntent(Intent intent) {
+        setIntent(intent);
+        handleIntent(intent);
+    }
+
+    private void handleIntent(Intent intent){
+        DbSearchAsyncTask dbSearchAsyncTask = new DbSearchAsyncTask(this, intent);
```

Comment:
Do the test on the intent to see if it is a search here instead of inside of the async task. There is no need to kick off an async task if it isn't needed.